datastore/src/main/java/com/ds/datastore/BookStoreController.java

```diff
@@ -3,78 +3,229 @@
+            store.setPhone(!jso.get("phone").isJsonNull() ? jso.get("phone").getAsString(): null);
+            store.setStreetAddress(!jso.get("streetAddress").isJsonNull() ? jso.get("streetAddress").getAsString(): null);
+            //Not including the List of books contained in the store
+            EntityModel<BookStore> entityModel = assembler.toModel(store);
+            entModelList.add(entityModel);
+        }
+        return CollectionModel.of(entModelList, linkTo(methodOn(BookStoreController.class).all()).withSelfRel());
     }
 
     @PutMapping("/bookstores/{storeID}")
     protected BookStore updateBookStore(@RequestBody BookStore newBookStore, @PathVariable Long storeID) {
-        return repository.findById(storeID)
+        return storeRepository.findById(storeID)
                 .map(bookStore -> {
-                    //TODO: add more stuff here
-                    return repository.save(bookStore);
+                    if(newBookStore.getName() != null) bookStore.setName(newBookStore.getName());
+                    if(newBookStore.getPhone() != null) bookStore.setPhone(newBookStore.getPhone());
+                    if(newBookStore.getStreetAddress() != null) bookStore.setStreetAddress(newBookStore.getStreetAddress());
+                    return storeRepository.save(bookStore);
                 })
                 .orElseThrow(() -> new BookStoreNotFoundException(storeID));
     }
 
-    @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/bookstores/{storeID}")
-    protected void deleteBookStore(@PathVariable Long storeID) {
-        repository.findById(storeID).orElseThrow(() -> new BookStoreNotFoundException(storeID));
-        repository.deleteById(storeID);
+    protected ResponseEntity deleteBookStore(@PathVariable Long storeID) throws Exception{
+        try{
+            storeRepository.findById(storeID).orElseThrow(() -> new BookStoreNotFoundException(storeID));
+            storeRepository.deleteById(storeID);
+            this.map.remove(storeID);
+            URL url = new URL("http://71.172.193.59:8080/hub/" + storeID);
+            HttpURLConnection con = (HttpURLConnection) url.openConnection();
+            con.setRequestMethod("DELETE");
+            con.setRequestProperty("Content-Type", "application/json");
+            con.setDoOutput(true);
+            con.connect();
+            int x = con.getResponseCode();
+            List<Book> books = bookRepository.findByStoreID(storeID);
+            for (Book book : books) {
+                bookRepository.delete(book);
+            }
+            return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+        }catch (BookStoreNotFoundException e){
+            if(this.map.containsKey(storeID)){
+                return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+            }else{
+                throw e;
+            }
+        }
+    }
+
+    @DeleteMapping("/bookstores")
```

Comment:
This makes sense - the hub calls this to to indicate to remove a bookstore from the mapping. Im thinking to maybe name the path `/bookstores/map` so that its clear? What do you think?